lib/sinon/assert.js

```diff
@@ -78,6 +78,11 @@ function mirrorPropAsAssertion(name, method, message) {
         var args = slice.call(arguments, 1);
         var failed = false;
 
+        // Hack to be replaced
```

Comment:
Lines 82-84 check for one of the sinon match objects, and if found, revert to using the original message formatting without colored diffs. I did this to avoid making a lot of changes in a single PR, and because the choice to print a diff for a sinon match comparison is a little bit more ambigious than comparing two plain objects (What do I print for `expect({x: true}).to.be.calledWith(match.falsy)`?). I was hoping one of the contributors could suggest a less hacky way to reserve the diff formating just for plain object comparison.